src/zcash/Note.hpp

```diff
@@ -116,6 +116,68 @@ class SproutNotePlaintext : public BaseNotePlaintext {
                                         ) const;
 };
 
+typedef std::pair<std::reference_wrapper<const SaplingEncCiphertext>, std::reference_wrapper<const SaplingNoteEncryption>> SaplingNotePlaintextEncryptionResult;
```

Comment:
I'm going to wait for @ebfull to push commits to this PR before making any changes, but thanks for taking an early look.  (and the non-const issue)

I'm using std::reference_wrapper to create a pair of references, rather than copying the SaplingEncCiphertext and SaplingNoteEncryption objects when creating the pair.  

However, I now think its safer to just copy the objects and avoid the potential for dangling references, when the underlying object falls out of scope.